src/gerente/gerente.service.ts

```diff
@@ -0,0 +1,137 @@
+    }
+
+    criarContaBancaria(tipo: TipoConta, idCliente: number): Cliente{
+        const clientes = this.lerClientes();
+        const cliente = clientes.find(cliente => cliente.id === idCliente);
+        const novaConta = {
+            tipo,
+            saldo: 0,
+            gerente: {
+                nome: "Mariana",
+                id: "1"
+
+            },
+            proprietario: {
+                nome: cliente.nome,
+                id: cliente.id
+            }
+        }
+        
+        return this.atualizarContaCliente(idCliente, novaConta)
+
+    }
+
+    atualizarContaCliente(id:number, conta: ContaBancaria): Cliente{
+        const clientes = this.lerClientes()
+        
+        const indiceCliente = clientes.findIndex(cliente => cliente.id === Number(id))
+        
+        if(clientes[indiceCliente]?.contas?.length){
+            clientes[indiceCliente].contas.push(conta)
+        }else{
+            clientes[indiceCliente].contas = [conta]
+        }
+
+        this.guardarClientes(clientes)
+        return clientes[indiceCliente]
+    }
+
+    excluirContaBancaria(idCliente: number, tipoConta: TipoConta): Cliente {
+        const clientes = this.lerClientes();
+        const indiceCliente = clientes.findIndex(cliente => cliente.id === Number(idCliente));
+    
+        const cliente = clientes[indiceCliente];
+        if (cliente.contas) {
+            cliente.contas = cliente.contas.filter(conta => conta.tipo !== tipoConta);
+        }
+    
+        this.guardarClientes(clientes);
+
+        let contas = this.lerContas();
+        contas = contas.filter(conta => !(conta.tipo === tipoConta && conta.proprietario.id === idCliente));
+        this.guardarContas(contas);
+    
+        return cliente;
+    }
+
+    alterarTipoConta(idCliente: number, tipoContaAtual: TipoConta, novoTipoConta: TipoConta): Cliente {
+        const clientes = this.lerClientes();
+
+        const indiceCliente = clientes.findIndex(cliente => cliente.id === Number(idCliente));
+
+        const cliente = clientes[indiceCliente];
+        const conta = cliente.contas?.find(conta => conta.tipo === tipoContaAtual);
+        console.log(conta)
+
+        conta.tipo = novoTipoConta;
+        this.guardarClientes(clientes);
+        return cliente;
+    }
+    
+
+    criarCliente(nome: string): Cliente{
+        const clientes = this.lerClientes()
+        const novoCliente = {
+            nome,
+            id: clientes.length > 0 ? clientes[clientes.length - 1].id + 1 : 1
+        }
+        clientes.push(novoCliente)
+        this.guardarClientes(clientes)
+        return novoCliente
+    }
+
+    buscarClientePeloId(id: number): Cliente{
+        const clientes = this.lerClientes()
+        const cliente = clientes.find(cliente => cliente.id === Number(id))
+
+        if(!cliente){
+            console.log(`O cliente com id ${id} não foi encontrado`)
+        }
+
+        return cliente
+    }
```

Comment:
Quando você está buscando um cliente pelo ID, o código pode acabar retornando **undefined** se o cliente não for encontrado e acredito que o **`undefined`** é um tanto ambíguo e pode se relacionar com infinitas situações.

Nesse caso, penso que o uso do **`null`** é bem mais adequado, porque ele indica a ausência de um valor. Quando você usa null, você está dizendo explicitamente "mana, tem valor aqui não.". Por outro lado, **undefined** geralmente significa que algo está indefinido, entende?

Pode parecer besteira, mas essas questões de semântica ajudam pra baralho, quando precisamos identificar o problema pelos logs.

Então dentro do if, ce poderia colocar algo como:
```typescript
if(!cliente){
   console.log(`O cliente com id ${id} não foi encontrado`)
   return null
}
```